javalin/src/main/java/io/javalin/http/JavalinServletHandler.kt

```diff
@@ -73,6 +76,10 @@ class JavalinServletHandler(
      * because we need a lazy evaluation of next tasks in a chain to support multiple concurrent stages.
      */
     internal fun queueNextTaskOrFinish() {
+        if (ctx.attribute<Boolean>("javalin-redirected") == true) {
+            ctx.attribute("javalin-redirected", null)
+            redirected = true
+        }
```

Comment:
I'm not sure if we even need the `redirected` property, we could just check it in `executeNextTask()` 🤔